Remove anchor links with nested button tags and improve styling of buttons

synicalsyntax opened the pull request:
As a followup to #12770 where @andersk pointed out that the root reason behind the linkified whitespace for the Change name button was that it was improperly nested inside an anchor link (invalid HTML), this PR includes a series of commits to fix #6126.

First, remove anchor links with nested buttons from user settings Handlebars templates (fb7dd5f)
  - Replace link styling with .sea-green class (a40bad7)

Now that the user settings buttons are no longer anchor links but buttons, improve their styling (along with other buttons in the codebase)
- Disabled buttons should not react on hover/focus/other user action for proper UX (7047c94)
- Improve coloring of buttons in night mode (border/text color for classes like sea-green and btn-warning was previously lost) (92ccc5a)

Refactor SCSS of portico buttons in preparation for transitioning anchor links with nested buttons to anchor links with just button styling (f8015d1 and 35b11aa)

Transition anchor links needing button styling to use the new `.button` class that inherits button styling (28043b3)

Remove unused CSS along the way (5f8b0e8 and 8df3662)

andersk commented:
Yay! Wondering why the `<button>` inside `<a>` HTML errors weren’t being caught in CI, I found that `PorticoDocumentationSpider` was treating even the listed portico URLs as external—fixed as #12813. If we merge this before #12813, we can remove the filters I added there to ignore these errors.

timabbott commented:
@synicalsyntax thanks for working on this!  I merged a few commits that had no issuse, and noticed a few issues:
* There's a few regressions I saw in testing.  In particular, The last two commits break the hover behavior on buttons like "See plans and pricing", as well as create this super weird white effect if you right-click on the "Download" buttons on /apps.
* Was it intentional to change the button colors to sea-green from blue for e.g. the "Change name" button?  @rishig what are your thoughts on this change?  While it's possible the color change is an improvement, it'd be a lot more convenient to review a change the implementation that wasn't mixed with potentially debateable styling changes.

synicalsyntax commented:
@timabbott What was the original hover effect of the portico buttons? I saw the same exact "weird" effect on zulipchat.com and thought it was intended

timabbott commented:
For me, zulipchat.com behaves differently from this branch

-Tim Abbott (mobile)

On Thu, Jul 18, 2019, 4:47 PM Cynthia Lin <notifications@github.com> wrote:

> @timabbott <https://github.com/timabbott> What was the original hover
> effect of the portico buttons? I saw the same exact "weird" effect on
> zulipchat.com and thought it was intended
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zulip/zulip/pull/12809?email_source=notifications&email_token=AAU6NWVSNEZL4OGBA4YRMRDQAD6JFA5CNFSM4IEW3U5KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2KEYBI#issuecomment-513035269>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAU6NWSLXWFOGMDOXMIXSALQAD6JFANCNFSM4IEW3U5A>
> .
>

rishig commented:
I think blue makes more sense, since these aren't really buttons.

Once we have a white background for the settings pages, I think we'll be able to remove the border.

synicalsyntax commented:
@timabbott I backed up the commits changing disabled button behavior to a local branch and will bring them back in a future PR.

I modified aa35f91a1dd76dddf9cf6d1e99911f0f2b7a81a4 so that the buttons maintain the blue link color that @rishig requested, but I left 2c93820f3 since the blue link color wouldn't show up in night mode if it weren't merged along with it; it doesn't have the improved border/text color changes from the original PR, just keeps the background color black and applying the border/text color changes **if** it has an extra class that modifies it. I can move it to the upcoming PR if you want to drop the commit from this PR, just let me know.

The weird white effect you noted for the portico buttons was due to the brightness filter not working properly on Firefox (button or anchor link regardless), so I just changed them to background-color changes (like they *should* be, since filter() is more meant for images not elements).

timabbott commented:
I merged the first two commits, thanks @synicalsyntax .  The last, for portico pages, seems to still have lost the hover behavior on this button: 

![image](https://user-images.githubusercontent.com/2746074/61569868-3bd4b200-aa3e-11e9-81c5-ea74a0f83da1.png)

(on localhost:9991/hello)

In a way that makes that button feel a lot less clickable than it used to be.

synicalsyntax commented:
@timabbott fixed 
![image](https://user-images.githubusercontent.com/15116870/61572504-86533e80-aad1-11e9-8cd6-865c5cd1b86a.png)

timabbott commented:
@rishig can you do a quick visual pass to check these important portico buttons/UI all look correct after this change?

rishig commented:
What pages does this affect? Checked /hello; all seems good there (though opened #12853 for some unrelated things I noticed).

synicalsyntax commented:
@rishig First commit affects /hello, /apps, and /plans (plans is hard to see in development because we redirect for some reason); second commit fixes regressions in buttons in the webapp when this PR was merged prematurely

timabbott commented:
It directly also impacts /plans and /apps.  Potentially also other portico portico things unexpectedly.  Probably audit of those can be rolled into auditing the styling changes of our default text font.

This lgtm, merged, thanks @synicalsyntax!

I also merged the series ending with 40e8fd24ffd088d463e7a61a3b5aefd1c8128c16 to make it easier to test /plans and /terms in development; @alexandraciobica FYI.
